sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignature.java

```diff
@@ -113,12 +117,148 @@ static Builder builder() {
     Method targetMethod();
   }
 
-  /** A type of optional parameter of the {@link DoFn.ProcessElement} method. */
-  public enum Parameter {
-    BOUNDED_WINDOW,
-    INPUT_PROVIDER,
-    OUTPUT_RECEIVER,
-    RESTRICTION_TRACKER
+  /** A descriptor for an optional parameter of the {@link DoFn.ProcessElement} method. */
```

Comment:
`of a {@link DoFn} method`?

In general these could be to any method within the `DoFn` reflection-magic, right?